src/transport.rs

```diff
@@ -63,6 +63,51 @@ pub enum Service {
     ReceivePack,
 }
 
+/// HTTP implementation related info for various services.
+///
+/// This information was pulled from
+/// <https://github.com/libgit2/libgit2/blob/2ecc8586f7eec4063b5da1563d0a33f9e9f9fcf7/src/libgit2/transports/http.c#L68-L95>.
+impl Service {
+    /// The HTTP Method used by libgit2's implementation for http(s) transport.
+    pub const fn http_method(self) -> http::Method {
+        use http::Method;
+
+        match self {
+            Service::UploadPackLs | Service::ReceivePackLs => Method::GET,
+            Service::UploadPack | Service::ReceivePack => Method::POST,
+        }
+    }
+
+    /// The value of the HTTP "Accept" header that is used by libgit2's implementation for http(s) transport.
+    pub const fn http_accept_header(self) -> &'static str {
+        match self {
+            Service::UploadPackLs => "application/x-git-upload-pack-advertisement",
```

Comment:
Is this a public interface exposed by libgit2? If not I tend not to add this. We don't really want to get into troubles that there is no stability guarantee in libgit2 while git2-rs exposes it.

Reply:
The entire `Service` type is native to `git2-rs` -- libgit2 doesn't have anything really matching it. The closest thing is https://github.com/libgit2/libgit2/blob/main/src/libgit2/transports/http.c but that is not publicly exposed. I can definitely understand not merging this because of that.

Reply:
Really appreciate your understanding. Yeah we need to weight on that.

Anyway, as GitHub keeps diff in a pull request indefinitely, this can be used as a reference when people need such a glue code, which is great!

Per discussion, I am going to close this and thank you again!

Reply:
Of course! You're welcome!